django/utils/html.py

```diff
@@ -294,20 +299,20 @@ def handle_word(self, word):
                 )
                 return mark_safe(f'{lead}{middle}{trail}')
             else:
-                if self.safe_input:
+                if safe_input:
                     return mark_safe(word)
-                elif self.autoescape:
+                elif autoescape:
                     return escape(word)
-        elif self.safe_input:
+        elif safe_input:
             return mark_safe(word)
-        elif self.autoescape:
+        elif autoescape:
             return escape(word)
         return word
 
-    def trim_url(self, x):
-        if self.trim_url_limit is None or len(x) <= self.trim_url_limit:
+    def trim_url(self, x, trim_url_limit):
```

Comment:
The `trim_url_` prefix for the new argument seems unnecessary here.
Also, if we're expecting it to be passed as a keyword, let's enforce that.

```suggestion
    def trim_url(self, x, *, limit=None):
```